perception/map_based_prediction/src/map_based_prediction_node.cpp

```diff
@@ -1076,23 +1091,42 @@ LaneletsData MapBasedPredictionNode::getCurrentLanelets(const TrackedObject & ob
     return {};
   }
 
-  LaneletsData closest_lanelets;
+  LaneletsData object_lanelets;
+  std::optional<std::pair<double, lanelet::Lanelet>> closest_lanelet{std::nullopt};
   for (const auto & lanelet : surrounding_lanelets) {
     // Check if the close lanelets meet the necessary condition for start lanelets and
-    // Check if similar lanelet is inside the closest lanelet
+    // Check if similar lanelet is inside the object lanelet
     if (
       !checkCloseLaneletCondition(lanelet, object, search_point) ||
-      isDuplicated(lanelet, closest_lanelets)) {
+      isDuplicated(lanelet, object_lanelets)) {
       continue;
     }
 
-    LaneletData closest_lanelet;
-    closest_lanelet.lanelet = lanelet.second;
-    closest_lanelet.probability = calculateLocalLikelihood(lanelet.second, object);
-    closest_lanelets.push_back(closest_lanelet);
+    // Memorize closest lanelet
+    // NOTE: The object may be outside the lanelet.
+    if (
+      lanelet.first < dist_threshold_for_searching_lanelet_ &&
+      (!closest_lanelet || lanelet.first < closest_lanelet->first)) {
+      closest_lanelet = lanelet;
+    }
```

Comment:
I think we need validation process to check object yaw and lanelet yaw diff is small enough so that we will not get the unexpected lanelet such as opposite lanelet

Reply:
We call checkCloseLaneletCondition before the above condition so we do not have to check the angle and even distance (Now I removed the distance check).